packages/store/ts/codegen/field.ts

```diff
@@ -137,6 +137,22 @@ export function renderFieldMethods(options: RenderTableOptions): string {
                 "uint256 _index",
               ])}) internal view returns (${portionData.typeWithLocation}) {
                 ${_keyTupleDefinition}
+
+                ${
+                  // If the index is within the static length,
+                  // but ahead of the dynamic length, return zero
+                  typeWrappingData && typeWrappingData.kind === "staticArray" && field.arrayElement
+                    ? `
+                uint256 _byteLength = ${_store}.getDynamicFieldLength(_tableId, _keyTuple, ${dynamicSchemaIndex});
+                uint256 dynamicLength = _byteLength / ${portionData.elementLength};
+                uint256 staticLength = ${typeWrappingData.staticLength};
+
+                if (_index < staticLength && _index >= dynamicLength) {
+                  return ${renderCastStaticBytesToType(field.arrayElement, `bytes${field.arrayElement.staticByteLength}(new bytes(0))`)};
+                }`
+                    : ``
+                }
```

Comment:
this is being passed through formatting afterwards, so we can adjust the spacing here to match the indentation of the source file (= add 7 spaces or something like that to lines 146-152 so the indentation matches the ``)